aspnet/fundamentals/caching/sample/test/CachingSample.Tests/MemoryCacheTests.cs

```diff
@@ -0,0 +1,97 @@
+﻿using System.Threading;
+using Microsoft.Extensions.Caching.Memory;
+using Microsoft.Extensions.Primitives;
+using Xunit;
+
+namespace CachingSample.Tests
+{
+    public class MemoryCacheTests
+    {
+        private readonly MemoryCache _memoryCache;
+        private readonly string _cacheItem = "value";
+        private readonly string _cacheKey = "key";
+        private string _result;
+
+        public MemoryCacheTests()
+        {
+            _memoryCache = new MemoryCache(new MemoryCacheOptions());
+        }
+
+        [Fact]
+        public void RemovalFiresCallback()
```

Comment:
```
            var pause = new ManualResetEvent(false);

            _memoryCache.Set(_cacheKey, _cacheItem,
                new MemoryCacheEntryOptions()
                .RegisterPostEvictionCallback(
                    (key, value, reason, substate) =>
                    {
                        _result = $"'{key}':'{value}' was evicted because: {reason}";
                        pause.Set();
                    }
                ));

            _memoryCache.Remove(_cacheKey);

            Assert.True(pause.WaitOne(500));

            Assert.Equal("'key':'value' was evicted because: Removed", _result);
```